File: meteorological_analysis_backend/src/main/java/com/example/meteorologicalAnalysis/constant/WeatherDataType.java
package com.example.meteorologicalAnalysis.constant;

import com.example.meteorologicalAnalysis.common.BusinessException;
import lombok.Getter;

@Getter
public enum WeatherDataType {
    TEMPERATURE("temperature", "气温"),
    DEW_POINT("dew_point", "露点气温"),
    PRESSURE("pressure", "气压"),
    WIND_DIRECTION("wind_direction", "风向"),
    WIND_SPEED("wind_speed", "风速");

    private final String column;
    private final String label;

    WeatherDataType(String column, String label) {
        this.column = column;
        this.label = label;
    }

    public static WeatherDataType fromColumn(String column) {
        for (WeatherDataType type : values()) {
            if (type.column.equalsIgnoreCase(column)) {
                return type;
            }
        }
        throw new BusinessException(ErrorCode.PARAMS_ERROR, "不支持的气象数据类型: " + column);
    }
}

File: meteorological_analysis_frontend/src/constant/WeatherDataType.ts
export enum WeatherDataType {
    TEMPERATURE = 'temperature',
    DEW_POINT = 'dew_point',
    PRESSURE = 'pressure',
    WIND_DIRECTION = 'wind_direction',
    WIND_SPEED = 'wind_speed',
}

export const WeatherDataTypeLabel: Record<WeatherDataType, string> = {
    [WeatherDataType.TEMPERATURE]: '气温',
    [WeatherDataType.DEW_POINT]: '露点气温',
    [WeatherDataType.PRESSURE]: '气压',
    [WeatherDataType.WIND_DIRECTION]: '风向',
    [WeatherDataType.WIND_SPEED]: '风速',
};

export function fromColumn(column: string): WeatherDataType {
    const type = Object.values(WeatherDataType).find(
        (type) => type.toLowerCase() === column.toLowerCase()
    );
    if (!type) {
        throw new Error(`不支持的气象数据类型: ${column}`);
    }
    return type;
}
